server/src/internalClusterTest/java/org/elasticsearch/health/HealthMetadataServiceIT.java

```diff
@@ -40,33 +48,43 @@ public void testEachMasterPublishesTheirThresholds() throws Exception {
             int numberOfNodes = 3;
             Map<String, String> watermarkByNode = new HashMap<>();
             Map<String, ByteSizeValue> maxHeadroomByNode = new HashMap<>();
+            Map<String, HealthMetadata.ShardLimits> shardLimitsPerNode = new HashMap<>();
             for (int i = 0; i < numberOfNodes; i++) {
                 ByteSizeValue randomBytes = ByteSizeValue.ofBytes(randomLongBetween(6, 19));
                 String customWatermark = percentageMode ? randomIntBetween(86, 94) + "%" : randomBytes.toString();
                 ByteSizeValue customMaxHeadroom = percentageMode ? randomBytes : ByteSizeValue.MINUS_ONE;
-                String nodeName = startNode(internalCluster, customWatermark, customMaxHeadroom.toString());
+                var customShardLimits = new HealthMetadata.ShardLimits(randomIntBetween(0, 1000), randomIntBetween(1001, 2000));
+                String nodeName = startNode(internalCluster, customWatermark, customMaxHeadroom.toString(), customShardLimits);
                 watermarkByNode.put(nodeName, customWatermark);
                 maxHeadroomByNode.put(nodeName, customMaxHeadroom);
+                shardLimitsPerNode.put(nodeName, customShardLimits);
+
             }
             ensureStableCluster(numberOfNodes);
 
             String electedMaster = internalCluster.getMasterName();
             {
-                HealthMetadata.Disk diskMetadata = HealthMetadata.getFromClusterState(internalCluster.clusterService().state())
-                    .getDiskMetadata();
+                var healthMetadata = HealthMetadata.getFromClusterState(internalCluster.clusterService().state());
+                var diskMetadata = healthMetadata.getDiskMetadata();
```

Comment:
We should leave these now, but for the future let's please have these cosmetics as a separate PR as it makes reviewing a bit murkier when they're baked in with business logic changes.

Reply:
sure!